bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java

```diff
@@ -108,10 +108,12 @@ public int size() {
 
     /**
      * Remove an item from the map.
+     *
      * @param key key of the item to be removed
+     * @return boolean to indicate whether given key is removed.
      */
-    public void remove(K key) {
-        map.remove(key);
+    public boolean remove(K key) {
+        return map.remove(key) != null;
```

Comment:
This will return `false`, even if the map has the key, but the value is null.
I think we should first check map.contains(), and return the result of that, after removing.

Reply:
+1